pkg/importer/spanner/sysl.arrai

```diff
@@ -0,0 +1,140 @@
+# size returns the size of an attribute
+let size = \length
+    cond {
+        length = 'MAX': '',
+        //seq.has_prefix('0x', length): $`(${//seq.trim_prefix('0x', length)})`,
+        length > 0: $`(${length})`,
+    };
+
+# hasAttributePattern checks whether an attribute has associated patterns
+let hasAttributePattern = \entity \attr
+    cond {
+        attr.options?:{}: true,
+        (entity.primary_key where //seq.contains(attr.name, .)): true,
+        let fks = (entity.foreign_keys => ((.foreign_keys where .attribute = attr.name) =>
+            .reference_attribute orderby .)) where .; fks: true,
+        attr.length = 'MAX': true,
+        //seq.has_prefix('0x', attr.length): true,
+    };
+
+# hasEntityPattern checks whether an entity has associated patterns
+let hasEntityPattern = \entity \model
+    entity.primary_key | entity.foreign_keys | entity.cluster | model.indexes;
+
+# attributePatternDelimiter returns the token based on attribute value
+let attributePatternDelimiter = \entity \attr \token
+    cond {
+        hasAttributePattern(entity, attr): token
+    };
+
+# entityPatternDelimiter returns the token based on entity value
+let entityPatternDelimiter = \entity \model \token
+    cond {
+        hasEntityPattern(entity, model): token
+    };
+
+# sortingOrder determines and appends sorting order
+let sortingOrder = \e \attr
+    let keyOrder = //seq.split('(', //seq.join('', (e.primary_key where //seq.contains(attr.name, .)) orderby .));
+    cond {
+        keyOrder count > 1: cond keyOrder(1) {
+            'asc)': ', ~asc',
+            'desc)': ', ~desc'
+        }
+    };
+
+# compareColumnOrder compares the col order between primary key and table
+let compareColumnOrder = \entity
+    let pk = (entity.primary_key orderby .) >> //seq.split('(', .) >> .(0);
+    cond {
+        pk & (entity.attributes >> .name): false,
+        _: true,
+    };
+
+# attributePattern generates the pattern constituents depending on the mode
+let attributePattern = \entity \attr \mode
+    cond {
+        mode = 0 && (entity.primary_key where //seq.contains(attr.name, .)): $`~pk${sortingOrder(entity, attr)}`,
+        let fks = (entity.foreign_keys => ((.foreign_keys where .attribute = attr.name) => .reference_attribute orderby .)) where .;
+        mode = 1 && fks: '~fk',
```

Comment:
Nit: don't need the `let`, and can short-circuit on the `mode` arg:
```suggestion
        mode = 1 && entity.foreign_keys => (.foreign_keys where .attribute = attr.name) => .reference_attribute where .: '~fk',
```
(untested)